infra/bisector.py

```diff
@@ -0,0 +1,170 @@
+  args = parser.parse_args()
+
+  error_sha = init_bisection(args.project_name, args.commit_old,
+                             args.commit_new, args.engine, args.sanitizer,
+                             args.architecture, args.test_case,
+                             args.fuzzer_name)
+  if not error_sha:
+    print('No error was found in commit range %s:%s' %
+          (args.commit_old, args.commit_new))
+  else:
+    print('Error was introduced at commit %s' % error_sha)
+
+
+def init_bisection(project_name, commit_old, commit_new, engine, sanitizer,
+                   architecture, test_case, fuzzer_name):
+  """Creates an bisection call that kicks off the error detection.
+
+  This function is necessary in order to get the error code of the newest commit. This
+  sets a standard for what the error actually is.
+
+  Args:
+    project_name: The name of the oss fuzz project that is being checked
+    commit_old: The oldest commit in the error regression range
+    commit_new: The newest commit in the error regression range
+    engine: The fuzzing engine to be used
+    sanitizer: The fuzzing sanitizer to be used
+    architecture: The system architecture being fuzzed
+    test_case: The file path of the test case that triggers the error
+    fuzzer_name: The name of the fuzzer to be tested
+
+  Returns:
+    The commit SHA that introduced the error or None
+  """
+  local_store_path = 'tmp'
+  repo_url = infer_main_repo(project_name, local_store_path, commit_old)
+  repo_manager = RepoManager(repo_url, local_store_path)
+  commit_list = repo_manager.get_commit_list(commit_old, commit_new)
+  build_fuzzer_from_commit(project_name, commit_list[0], repo_manager.repo_dir,
+                           engine, sanitizer, architecture, repo_manager)
+  error_code = reproduce_impl(project_name, fuzzer_name, False, [], [],
+                              test_case)
+  # Handle the case where there is only one SHA passed in
+  if len(commit_list) == 1:
+    if not error_code:
+      return None
+    return commit_list[0]
+  index = bisection(project_name, 0,
+                    len(commit_list) - 1, commit_list, repo_manager,
+                    len(commit_list), error_code, engine, sanitizer,
+                    architecture, test_case, fuzzer_name)
+  if index is not None:
+    return commit_list[index]
+  return None
+
+
+def bisection(project_name, commit_new_idx, commit_old_idx, commit_list,
+              repo_manager, last_error, error_code, engine, sanitizer,
+              architecture, test_case, fuzzer_name):
+  """Returns the commit ID where a bug was introduced.
+
+  Args:
+    project_name: The name of the oss fuzz project being tested
+    commit_old_idx: The oldest commit SHA index in the search space
+    commit_new_idx: The newest commit SHA index in the search space
+    commit_list: The list of all commit SHAs
+    repo_manager: The class handling all of the git repo calls
+    last_error: The index where the last error was found
```

Comment:
`last_error_idx` or maybe `last_reproducible_idx`, otherwise it seems like this variable represents an error somehow